src/ripple/protocol/STArray.h

```diff
@@ -71,11 +71,35 @@ class STArray final
         return emplace(n, buf, std::move(*this));
     }
 
-    void push_back (const STObject & object)
+    STObject& operator[] (std::size_t j)
+    {
+        return v_[j];
+    }
+
+    STObject const& operator[] (std::size_t j) const
+    {
+        return v_[j];
+    }
+
+    STObject& back() { return v_.back(); }
```

Comment:
@vinniefalco: you give an excellent symmetry argument.  There is no `push_front()`, and I wouldn't ague for one, due to the efficiency issues you point out.  So, yes, there should probably not be a `front()`.